src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -166,4 +176,18 @@ public void invalidateUserJwt(final Cookie customJWTCookie, final HttpServletRes
         thirdPartyAuthToken.setMaxAge(0);
         response.addCookie(thirdPartyAuthToken);
     }
+
+    public void validateRoles(List<Role> dbRoles, String roles) {
+        final List<String> rolesList = removeSquareBracketsAndTrim(Arrays.asList(roles.split(",")));
+        final List<String> dbRolesList = dbRoles.stream().map(role -> roleMapper.roleToRoleDto(role).getName()).collect(Collectors.toList());
+
+        if(!new HashSet<>(dbRolesList).containsAll(rolesList)){
```

Comment:
Can you extract the hashset instantiation out for readability?

HashSet<String> userStoredRoles = new HashSet<>(dbRolesList);

Reply:
Using a Set<String> for this now